eZ/Publish/Core/Persistence/Legacy/Tests/UserPreference/Gateway/DoctrineDatabaseTest.php

```diff
@@ -0,0 +1,141 @@
+     */
+    public function testInsert()
+    {
+        $id = $this->getGateway()->setUserPreference(new UserPreferenceSetStruct([
+            'userId' => 14,
+            'name' => 'setting_3',
+            'value' => 'value_3',
+        ]));
+
+        $data = $this->loadUserPreference($id);
+
+        $this->assertEquals([
+            'id' => $id,
+            'user_id' => '14',
+            'name' => 'setting_3',
+            'value' => 'value_3',
+        ], $data);
+    }
+
+    /**
+     * @covers \eZ\Publish\Core\Persistence\Legacy\UserPreference\Gateway\DoctrineDatabase::setUserPreference()
+     */
+    public function testUpdateUserPreference()
+    {
+        $userPreference = new UserPreferenceSetStruct([
+            'userId' => 14,
+            'name' => 'timezone',
+            'value' => 'Europe/Warsaw',
+        ]);
+
+        $this->getGateway()->setUserPreference($userPreference);
+
+        $this->assertEquals([
+            'id' => (string) self::EXISTING_USER_PREFERENCE_ID,
+            'user_id' => '14',
+            'name' => 'timezone',
+            'value' => 'Europe/Warsaw',
+        ], $this->loadUserPreference(self::EXISTING_USER_PREFERENCE_ID));
+    }
+
+    /**
+     * @covers \eZ\Publish\Core\Persistence\Legacy\UserPreference\Gateway\DoctrineDatabase::countUserPreferences()
+     */
+    public function testCountUserPreferences()
+    {
+        $this->assertEquals(3, $this->getGateway()->countUserPreferences(
+            self::EXISTING_USER_PREFERENCE_DATA['user_id']
+        ));
+    }
+
+    /**
+     * @covers \eZ\Publish\Core\Persistence\Legacy\UserPreference\Gateway\DoctrineDatabase::loadUserPreferences()
+     */
+    public function testLoadUserPreferences()
+    {
+        $userId = 14;
+        $offset = 1;
+        $limit = 2;
+
+        $results = $this->getGateway()->loadUserPreferences($userId, $offset, $limit);
+
+        $this->assertEquals([
+            [
+                'id' => '2',
+                'user_id' => '14',
+                'name' => 'setting_1',
+                'value' => 'value_1',
+            ],
+            [
+                'id' => '3',
+                'user_id' => '14',
+                'name' => 'setting_2',
+                'value' => 'value_2',
+            ],
+        ], $results);
+    }
+
+    /**
+     * Return a ready to test DoctrineStorage gateway.
+     *
+     * @return \eZ\Publish\Core\Persistence\Legacy\UserPreference\Gateway\DoctrineDatabase
+     */
+    protected function getGateway(): DoctrineDatabase
+    {
+        return new DoctrineDatabase(
+            $this->getDatabaseHandler()->getConnection()
+        );
+    }
+
+    /**
+     * @param int $id
+     *
+     * @return array
+     *
+     * @throws \Doctrine\DBAL\DBALException
+     */
+    private function loadUserPreference(int $id): array
+    {
+        $data = $this->connection
+            ->executeQuery('SELECT * FROM ezpreferences WHERE id = :id', ['id' => $id])
```

Comment:
Please use Doctrine Query Builder.
```php
$queryBuilder = $this->connection->createQueryBuilder();
$queryBuilder
    ->select('id', 'user_id', 'name', 'value')
    ->from('ezpreferences', 'p')
    ->where(
        $queryBuilder->expr()->eq(
            'p.id', 
            $queryBuilder->createPositionalParameter($id, ParameterType::INTEGER)
      )
;

return $queryBuilder->execute()->fetchAll(FechMode:::ASSOCIATIVE);
```
^ written in comment textarea, so can contain mistakes ;)